### PR TITLE
DIP-32: Email Welcome Packs with Testnet Tokens for Workshops at DEVCon(nect)

### PR DESCRIPTION
**Pull request by 0xkkonrad**
## Checklist
 * [x ] Correct markdown (e.g. headers with ##, header surrounded by ---)
 * [ x] syntax check
 * [x ] filename is DIP-X.md (with X being the correct DIP number)
 * [ x] header contains (in this order)
   * [x ] DIP
   * [x ] Title
   * [x ] Status
   * [x ] Themes
   * [x ] Tags
   * [x ] Authors
   * [x ] Resources Required (optional)
   * [x ] Discussion (with link to devcon forum)
   * [x ] Created (in the format year-month-day e.g 2020-4-20)
 * [ x] contains these sections:
   * [x ] Abstract
   * [ x] Rationale
   * [ x] Implementation
   * [ x] Operational Requirements & Ownership
 * [ x] PR title prefixed with DIP-X

### CODE REVIEW

File: DIPs/DIP-32.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+Discussion: https://forum.devcon.org/t/rfc-dip-for-simplifying-testnet-token-distribution-for-workshops-at-devcon-nect/3233/
+Created: 2023-9-29
+---
+
+# DIP for Email Welcome Packs with Testnet Tokens for Workshops at DEVCon(nect)
+
+Authors: kkonrad.eth; hugo0.eth. We would like to thank Ligi for the initial inspiration at EthBerlin3 and idea for DIP-32.
+
+---
+
```
Comment (ligi) [inline DIPs/DIP-32.md:18]:
```suggestion
```

File: DIPs/DIP-32.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+## Specs, Operational Requirements & Ownership
+
+**Proposal:** Distribute testnet tokens by including a link in each registration confirmation email.
+
+**Ownership:**
+The team behind Peanut Protocol can either own the entire process, which would require data sharing and is undesirable from a data handling perspective, or only implement the technology and thus own:
+1. creating the Pretix Plugin and implementing the Peanut Proctocol SDK in that. 
+2. Testing said plugin on a fake test event.
+3. Maintain the repo.
+4. Helping the DevCon(nect) temas use the new technology.
```
Comment (ligi) [inline DIPs/DIP-32.md:44]:
```suggestion
4. Helping the DevCon(nect) teams use the new technology.
```

**Review by ligi — CHANGES_REQUESTED** [review]
Thanks for the proposal - just some small nits - formally only the Rationale is missing
We will discuss it later this week (Thursday)

**Comment by vb7401** [comment]
It may also be worth a collaboration with emailwallet.org team, a group funded by EF's PSE! This way the ticket email can directly be the receiver of the test token.

**Comment by ligi** [comment]
@vb7401 can you elaborate how the collaboration would work on a technical level?

**Comment by vb7401** [comment]
I'm not a full expert, just a friend/fan of that team haha. From my understanding, they are building a smart contract wallet for arbitrary email addresses `email@domain.com`, where transferring tokens is gated by sending an email from `email@domain.com` with the specified action. This works because each email comes with a DKIM signature from the mailserver, which can then be verified on-chain cheaply using a ZKP, from which the action specified (sending money, sending an NFT) can be authorized.

It may be too early stage for them to support the specific types of test tokens this proposal needs, but roughly I am imagining that these tokens can directly be given to the smart contract wallets for each of the pretix emails instead of doing a flow with a claim link + sending them to another wallet. And in general, I am very bullish on emailwallet.org long-term being the right medium for "token airdrop to ticket holder" solutions, especially in communities that are less web3 focused/pilled!

**Comment by ligi** [comment]
I would need to dig into email-wallet from the surface I have concerns wrt centralisation and readyness - just wanted to try it out and got this:
image

So I think for DEVConnect we should go with peanut

**Comment by vb7401** [comment]
Yep, they're working on shipping a lot of the functionality for Devconnect, so it might be too difficult to make everything work out. Perhaps for a future Devcon 😃🤞

**Comment by ligi** [comment]
Sorry but we went with the PCDPass faucet solution in the end: https://twitter.com/_pk910_/status/1721472568765345955

Your solution is great - but we want to encourage users to use PCDPass

**Pull request closed without merging**